### PR TITLE
Remove o-filterable-list

### PR DESCRIPTION
**Pull request by anselmbradford**
@chosak was working with the Filterable List and we realized that there's a lot of validation infrastructure that appears unused.

There are two organisms, FilterableList and FilterableListControls. FilterableList contains FilterableListControls and also has a notification and results block. Neither of these blocks have styling and the purpose of FilterableList is to handle routing client-side validation errors from the FilterableListControls to its sibling notification. 

The validation appears to have been written primarily for date input fields, before the date input type gained widespread browser support. It also may have been written at a time when filterable list controls were thought to contain a much wider variety of inputs.

@chosak and I were unable to actually produce a state that shows a client-side validation error, since the fields are also validated on the server-side. Removing all the client-side validation removes a lot of unused code and complexity.

## Removals

- Remove `o-filterable-list`, `o-filterable-list_notification`, and `o-filterable-list_results` classes.
- Remove client-side validation of the filterable list controls inputs, deferring instead to browser-native checks and server-side checks.
- Remove all but the `email` validator, which is used by the email popup (nudgy guy).

## Changes

- Remove `error-messages-config.js` and move its relevant messages to the places they are used.
- Update Cypress helpers that targeted deleted classes to use `data-cy` attributes instead.

## How to test this PR

1. http://localhost:8000/about-us/blog/ should function as before and it should not be possible to get in a state without a notification showing.

### CODE REVIEW

File: cfgov/unprocessed/js/modules/util/validators.js [hunk 1]
```diff
@@ -5,55 +5,15 @@
...
+  EMAIL: {
+    INVALID: 'You have entered an invalid email address.',
+    INVALID_ES: 'La dirección de correo electrónico introducida no es válida.',
+    REQUIRED: 'Please enter an email address.',
+    REQUIRED_ES: 'Por favor, introduzca una dirección de correo electrónico.',
+  },
+};
 
 /**
  * email Determines if a field contains a email date.
```
Comment (chosak) [inline cfgov/unprocessed/js/modules/util/validators.js:19]:
```suggestion
 * email Determines if a field contains an email address.
```

File: cfgov/unprocessed/js/organisms/FilterableListControls.js [hunk 2]
```diff
@@ -275,5 +144,7 @@ function FilterableListControls(element) {
 }
 
 FilterableListControls.BASE_CLASS = BASE_CLASS;
```
Comment (chosak) [inline cfgov/unprocessed/js/organisms/FilterableListControls.js:146]:
```suggestion
```

It seems like this line is no longer needed? It was previously used [here](https://github.com/cfpb/consumerfinance.gov/blob/defb5e59415a10a1ec176a47f15a1cb4c8925199/cfgov/unprocessed/js/organisms/FilterableList.js#L35).

File: cfgov/unprocessed/js/organisms/FormSubmit.js [hunk 3]
```diff
@@ -5,11 +5,16 @@ import {
   BaseTransition,
   EventObserver,
 } from '@cfpb/cfpb-atomic-component';
-import ERROR_MESSAGES from '../config/error-messages-config.js';
 import Notification from '../molecules/Notification.js';
 import { scrollIntoView } from '../modules/util/scroll.js';
 
-const FORM_MESSAGES = ERROR_MESSAGES.FORM.SUBMISSION;
+const FORM_MESSAGES = {
```
Comment (chosak) [inline cfgov/unprocessed/js/organisms/FormSubmit.js:11]: Just to clarify: these aren't used any longer with FLs, right, after this PR? The error messages should all come from the backend, [for example](https://www.consumerfinance.gov/about-us/blog/?title=&topics=ans&language=ans&from_date=&to_date=).

**Review by chosak — APPROVED** [review]
This is great! Just a few minor comments.

File: cfgov/unprocessed/js/organisms/FormSubmit.js (hunk 3 above)
Comment (anselmbradford) [reply cfgov/unprocessed/js/organisms/FormSubmit.js:11]: FormSubmit is used in the email signup and popup (nudgy guy), which still does validation on the client-side.

While looking at this, I added a minor commit to standardize the validator imports https://github.com/cfpb/consumerfinance.gov/pull/7770/commits/be94a9e70ee122d4d1a3b7fa7d9b4cac1d37e195

File: cfgov/unprocessed/js/organisms/FilterableListControls.js (hunk 2 above)
Comment (anselmbradford) [reply cfgov/unprocessed/js/organisms/FilterableListControls.js:146]: This follows our general component patterns, so I think we should keep it to keep the component boilerplate the same:
https://github.com/cfpb/design-system/blob/main/packages/cfpb-forms/src/organisms/Multiselect.js#L578

This just allows retrieving the CSS class name from the component without needing to hardcode the CSS class name outside of the component.

**Review by wpears — APPROVED** [review]
coooooooooool

**Pull request merged**